src/screens/AddWalletSheet.tsx

```diff
@@ -0,0 +1,283 @@
+import { AddWalletList } from '@/components/add-wallet/AddWalletList';
+import { AddWalletItem } from '@/components/add-wallet/AddWalletRow';
+import { Box, globalColors, Inset, Stack, Text } from '@/design-system';
+import { useNavigation } from '@/navigation';
+import Routes from '@/navigation/routesNames';
+import React, { useCallback, useRef } from 'react';
+import * as i18n from '@/languages';
+import { HARDWARE_WALLETS, PROFILES, useExperimentalFlag } from '@/config';
+import { analytics, analyticsV2 } from '@/analytics';
+import { InteractionManager } from 'react-native';
+import { createAccountForWallet, walletsLoadState } from '@/redux/wallets';
+import WalletBackupTypes from '@/helpers/walletBackupTypes';
+import { createWallet } from '@/model/wallet';
+import WalletTypes from '@/helpers/walletTypes';
+import logger from '@/utils/logger';
+import { captureException } from '@sentry/react-native';
+import { useDispatch } from 'react-redux';
+import { backupUserDataIntoCloud } from '../handlers/cloudBackup';
+import showWalletErrorAlert from '@/helpers/support';
+import { WalletLoadingStates } from '@/helpers/walletLoadingStates';
+import { useWallets } from '@/hooks';
+
+const TRANSLATIONS = i18n.l.wallet.new.add_wallet;
+
+export const AddWalletSheet = () => {
+  const { goBack, navigate } = useNavigation();
+
+  const hardwareWalletsEnabled = useExperimentalFlag(HARDWARE_WALLETS);
+  const profilesEnabled = useExperimentalFlag(PROFILES);
+  const dispatch = useDispatch();
+  const creatingWallet = useRef<boolean>();
+  const {
+    isDamaged,
+    selectedWallet,
+    setIsWalletLoading,
+    wallets,
+  } = useWallets();
+
+  const onPressCreate = useCallback(async () => {
+    try {
+      analyticsV2.track(analyticsV2.event.addWalletFlowStarted, {
+        isFirstWallet: false,
+        type: 'new',
+      });
+      analytics.track('Tapped "Create a new wallet"');
+      if (creatingWallet.current) return;
+      creatingWallet.current = true;
+
+      // Show naming modal
+      InteractionManager.runAfterInteractions(() => {
+        goBack();
+      });
+      InteractionManager.runAfterInteractions(() => {
+        setTimeout(() => {
+          navigate(Routes.MODAL_SCREEN, {
+            actionType: 'Create',
+            asset: [],
+            isNewProfile: true,
+            onCancel: () => {
+              creatingWallet.current = false;
+              setIsWalletLoading(null);
+            },
+            onCloseModal: async (args: any) => {
+              if (args) {
```

Comment:
adding a note that none of this is new logic here

Reply:
🥭